jax_distribution/wrappers/jumanji.py

```diff
@@ -88,6 +89,72 @@ def step(
         return state, timestep
 
 
+class AgentIDWrapper(Wrapper):
+    """Add onehot agent IDs to observation."""
+
+    def __init__(self, env: Environment, has_global_state: bool = False):
+        super().__init__(env)
+        self.num_obs_features = self._env.num_obs_features + self._env.num_agents
+        self.has_global_state = has_global_state
+
+    def reset(self, key: chex.PRNGKey) -> Tuple[State, TimeStep]:
+        """Reset the environment."""
+        state, timestep = self._env.reset(key)
+
+        agent_ids = jnp.eye(self._env.num_agents)
+        new_agents_view = jnp.concatenate([agent_ids, timestep.observation.agents_view], axis=-1)
+
+        if self.has_global_state:
+            timestep.observation = ObservationGlobalState(
+                agents_view=new_agents_view,
+                action_mask=timestep.observation.action_mask,
+                step_count=timestep.observation.step_count,
+                global_state=timestep.observation.global_state,
+            )
+
+        else:
+            timestep.observation = Observation(
+                agents_view=new_agents_view,
+                action_mask=timestep.observation.action_mask,
+                step_count=timestep.observation.step_count,
+            )
```

Comment:
Instead of the if/else block could you not just do: `timestep.observation.replace(agents_view=new_agents_view)`? Might be `_replace` for `NamedTuple`, I can never remember

Reply:
the issue with _replace it consumes time more than creating new Observation so maybe this is one of @RuanJohn  reasons

Reply:
That is exactly the reason. For some reason replacing was slower than just recreating the observation.
I am not sure why this is the case. I can test it again on this PR.

Reply:
After a chat, we've decided to test it after the release